### ISSUE TITLE
can not run Nvidia-docker or docker with "--gpus"

### ISSUE THREAD

**Issue by guanshuidong** [NVIDIA/nvidia-container-toolkit#302]
After an accidental outage (when the docker with Nvidia was running), I restart the  Server Machine  and found the docker I ran was dismissed then I encounter problems when starting new docker:
if I run this :
`
sudo nvidia-docker run -it -p 2222:22 --name fedthe -d fcd9cd8b7603 /bin/bash
`,
the response is
`
docker: Error response from daemon: Unknown runtime specified nvidia.
See 'docker run --help'.
`
and no container was established.

If I run this:
`
sudo docker run --gpus all -it -p 2222:22 --name fedthe -d fcd9cd8b7603 /bin/bash
`
the response is
`
e711025175dc2bea0a64298c00062710c97eac4327cde3b302c58dfd6e583fe2
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy'
nvidia-container-cli: initialization error: load library failed: libnvidia-ml.so.1: cannot open shared object file: no such file or directory: unknown.
`
and one container was established but without GPU enabled.

I try to add /usr/lib/x86_64-linux-gnu/ to PATH ,and the result is the same.Thank you very much for helping me!

**Comment by guanshuidong** [comment]
well ,I  reinstall Nvidia-container-tool. and problem solved although I did not figure it out

**Comment by bawee** [comment]
Hello i'm getting the same load library failed error as the user above when running a nextflow pipeline with docker that uses the nvidia-runtime-toolkit. It seems that the error is only present in the new version released yesterday (1.14.4) but works on an identical computer running version 1.14.3. 

`Command error:
  docker: Error response from daemon: failed to create task for container: failed to create a shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy'
  nvidia-container-cli: initialization error: load library failed: libnvidia-ml.so.1: cannot open shared object file: no such file or directory: unknown.
  time="2024-01-24T14:05:51Z" level=error msg="error waiting for container: "`
  
  nvidia-runtime-toolkit. was installed using apt on Ubuntu 22.04.3 following instructions from https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html
  
  Thanks in advance!

**Comment by hopisaurus** [comment]
> Hello i'm getting the same load library failed error as the user above when running a nextflow pipeline with docker that uses the nvidia-runtime-toolkit. It seems that the error is only present in the new version released yesterday (1.14.4) but works on an identical computer running version 1.14.3.
> 
> `Command error: docker: Error response from daemon: failed to create task for container: failed to create a shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy' nvidia-container-cli: initialization error: load library failed: libnvidia-ml.so.1: cannot open shared object file: no such file or directory: unknown. time="2024-01-24T14:05:51Z" level=error msg="error waiting for container: "`
> 
> nvidia-runtime-toolkit. was installed using apt on Ubuntu 22.04.3 following instructions from https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html
> 
> Thanks in advance!

Try running a different image, it seems they deprecated using `latest` tag versioning. 

eg. 
```
docker run --gpus all nvidia/cuda:12.3.1-runtime-ubuntu22.04 nvidia-smi --query-gpu=uuid --format=csv
```

**Comment by elezar** [comment]
@guanshuidong in order to use the `nvidia` runtime when running docker, you need to configure it. This is usually done by adding it to the `/etc/docker/daemon.json` file. We also provide functionality in the `nvidia-ctk` CLI to do so. See our documentation [here](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html#configuring-docker).

Also, since a working NVIDIA driver is a prerequisite for using GPUs in a container, could you confirm that you are able to run `nvidia-smi` on your host?

Not that another thing to note is that the NVIDIA Container Runtime uses the `NVIDIA_VISIBLE_DEVICES` envvar in the container to determine which devices should be made available. This means that if the image you used `fedthe` does not set this envvar.

@bawee is this related to #305 where the issue was a non-NVIDIA driver installed on the system?
